MediaRecorder.bs

```diff
@@ -286,11 +276,35 @@ interface MediaRecorder : EventTarget {
     <li>Set |recorder|'s {{state}} to {{recording}}, and run the following steps
     in parallel:
     <ol>
-      <li>Start recording all tracks in |tracks| using the {{MediaRecorder}}'s
-      current configuration, and gather the data into a {{Blob}} <var>blob</var>
-      and queue a task, using the DOM manipulation task source, to
-      <a>fire an event</a> named <a>start</a> at <var>recorder</var>, then return
-      <code>undefined</code>.</li>
+      <li>Let the User Agent specify the |recorder|'s configuration for
+      recording all the tracks in |tracks|, given its current constraints. If
+      track data is available, the User Agent MAY take the content into account
+      when making this decision as long as this does not violate how the
+      configuration has previously been constrained (see earlier steps).</li>
+      <div class="note">
+        By looking at the content of the tracks when recording start, the User
+        Agent could take frame information into account. For example, if the
+        MediaStreamTrack is a remote track sourced from an RTCPeerConnection,
+        the User Agent is allowed to pick the same codec that the source RTP
+        stream uses, which may allow the User Agent to avoid the cost of
+        re-encoding by recording the raw encoded frames.
+      </div>
+      <li>Start recording all tracks in |tracks| using the |recorder|'s current
+      configuration and gather the data into a {{Blob}} <var>blob</var>. As soon
```

Comment:
Formatting: At "As soon as", are you still in the same execution cycle?
If "recording starts" doesn't happen in the same execution cycle, it would be clearer to take this step and what follows out of this numbered list and start a new one.

Reply:
We're already in "in parallel" steps, so queuing a task will happen in the next execution cycle. The "as soon as" may be superflous given that we've already said to start the encoding but I wanted to be clear that "onstart" happens as a result of starting to encode. But I removed the "as soon as" in case that was confusing.